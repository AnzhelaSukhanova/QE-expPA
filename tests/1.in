(set-option :produce-models true)
(set-logic BV)
(declare-fun y () (_ BitVec 4))
(assert
 (exists ((x (_ BitVec 4))) (or (bvult x y) (= x y))))
