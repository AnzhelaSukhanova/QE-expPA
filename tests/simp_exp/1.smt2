(set-option :produce-models true)
(set-logic BV)
(declare-fun y () (_ BitVec 4))
(assert
 (exists ((x (_ BitVec 4))) (or (bvult (bvshl #b0001 x) y) (= (bvshl #b0001 x) y))))
