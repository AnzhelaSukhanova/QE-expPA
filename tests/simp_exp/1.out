(let (($e1 (= y #b0000))) (let (($e2 (ite (bvult y #b1000) (ite (bvult y #b0100) (ite (bvult y #b0010) (ite $e1 #b1111 #b0000) #b0001) #b0010) #b0011))) (let (($e3 (= y #b0001))) (let (($e4 (= y #b0010))) (let (($e5 (= y #b0100))) (let (($e6 (= y #b1000))) (let (($e7 (bvult #b0001 y))) (let (($e8 (bvult #b0010 y))) (let (($e9 (bvult #b0100 y))) (let (($e10 (bvult #b1000 y))) (not (and (not $e8) (not $e7) (not $e9) (not $e4) (not $e3) (not $e10) (not $e5) $e1 (not $e6) (not $e7) (not $e1) (not $e8) (not $e3) (not $e9) (not $e4) (not $e10) (not $e5) $e1 (not $e6) (not $e7) (not $e1) (not (bvult (bvshl #b0001 (bvadd #b0001 $e2)) y)) (not (bvult (bvshl #b0001 $e2) y)) (not $e8) (not $e3) (not $e4)))))))))))))
